ephios/core/signup/methods.py

```diff
@@ -92,15 +107,51 @@ def participation_for(self, shift):
         except LocalParticipation.DoesNotExist:
             return None
 
+    def reverse_signup_action(self, shift):
+        return reverse("core:signup_action", kwargs=dict(pk=shift.pk))
+
+    def reverse_event_detail(self, event):
+        return event.get_absolute_url()
+
 
 class ParticipationError(ValidationError):
     pass
 
 
+def prevent_getting_participant_from_request_user(request):
+    original_user = request.user
+
+    class ProtectedUser(SimpleLazyObject):
+        def as_participant(self):
+            raise Exception("Access of request.user.as_participant in SignupViews is not allowed.")
```

Comment:
I tried all kinds of mechanics to enforce this in some kind of way, trying context managers, overwriting instance properties. This is what worked best. 
Of course we can replace the exception with a warning, but it works so far.